src/game_api/render_api.cpp

```diff
@@ -142,6 +143,197 @@ const char** RenderAPI::load_texture(std::string file_name)
     return load_texture(render_api, &file_name, 1);
 }
 
+using VanillaRenderHudFun = void(size_t, float, float, size_t);
+VanillaRenderHudFun* g_render_hud_trampoline{nullptr};
+void render_hud(size_t hud_data, float y, float opacity, size_t hud_data2)
+{
+    // hud_data and hud_data2 are the same pointer, but the second one is actually used (displays garbage if not passed)
+    LuaBackend::for_each_backend(
+        [&](LuaBackend& backend)
+        {
+            backend.pre_render_hud();
+            return true;
+        });
+
+    g_render_hud_trampoline(hud_data, y, opacity, hud_data2);
+
+    LuaBackend::for_each_backend(
+        [&](LuaBackend& backend)
+        {
+            backend.post_render_hud();
+            return true;
+        });
+}
+
+using VanillaRenderPauseMenuFun = void(float*);
+VanillaRenderPauseMenuFun* g_render_pause_menu_trampoline{nullptr};
+void render_pause_menu(float* drawing_info)
+{
+    LuaBackend::for_each_backend(

```

Comment:
Maybe move this to events.hpp/.cpp because of compile times? lua_backend.hpp is quite heavy.

Reply:
I did this, but the stuff I saw in events.cpp was more similar to the functions in render_api.cpp, not lua_backend.cpp. I did reduce the amount of functions in lua_backend.cpp from six to two though